### PR TITLE
Add setup and usage documentation to README

### LINKED ISSUE #5: Add documentation on how to use the font
Add some general documentation on how to use the font, in a _Getting started_-style, to the README. If more documentation is needed I propose putting it in a new `docs/` folder.

### PR DESCRIPTION
**Pull request by mondeja**
Closes #5

- Setup instructions with example using the font from CDN.
- Setup instructions with example downloading the font from NPM registry. The URL has the version hardcoded, this must be updated by release process or find a way to reference the latest version.
- Installation for module bundlers.
- Basic description of CSS classes.
- Improved badges images and README Markdown markup.

### CODE REVIEW

File: README.md
Comment (ericcornelissen) [inline README.md:26]: I would prefer a simpler style similar to [the README of the main project](https://github.com/simple-icons/simple-icons#readme).

Also I think we should include the version here as well, as we do in the main project.

```suggestion
```html
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/simple-icons-font@v2/font/simple-icons.min.css" type="text/css">
```
```

File: README.md
Comment (ericcornelissen) [inline README.md:32]: Could you keep the formatting style that is used in the main repository and write this as a single line? Also, using the text "here" for a link is bad for accessibility and often unnecessary, e.g here you can just do

```suggestion
1. [Download the latest version of the package][npm-registry-tarball-link], extract it, and move the content of the `font` folder to your desired location.
```

<sup>Alternatively, since this is technically invalid MarkDown, the two alternative lines should be indented with 3 spaces to line up properly.</sup>

File: README.md
Comment (ericcornelissen) [inline README.md:44]: I would reduce this down to just the `<i>` tag and move it to the "Usage" section, as this is how you use it regardless of how you set it up.

```suggestion
    <i class="simpleicons simpleicons-simpleicons simpleicons--color"></i>
</html>
```

File: README.md
Comment (ericcornelissen) [inline README.md:39]: i would only leave this part of the sample code in this section, and add that you may need to configure the path (and that you likely want it to be "absolute"):

```suggestion
    <link rel="stylesheet" href="/path/to/simple-icons.min.css" type="text/css">
```

File: README.md
Comment (ericcornelissen) [inline README.md:47]: Would prefer "Node Usage" here as that is what we use in the main repo. Also, the only relevant thing here is that you install it using NPM and use NodeJS to get it where you want. Whether you use bundlers or a script or something else entirely (including serving directly from `node_modules` :nauseated_face:) is up to the end user.

File: README.md
Comment (ericcornelissen) [inline README.md:50]: You didn't change this, but to be inline with the main repo I think we should have

```suggestion
$ npm install simple-icons-font
```

File: README.md
Comment (ericcornelissen) [inline README.md:76]: I would really be against having to update this for *every* release rather than every *major* release. Can we link to a page with all tarballs instead?

File: README.md
Comment (ericcornelissen) [inline README.md:62]: For consistency with the main repo, can we use square brackets here?

```suggestion
- `simpleicons-[ICON SLUG]`: Replace here `[ICON SLUG]` by with the
```

File: README.md
Comment (ericcornelissen) [inline README.md:28]: I would say this is the least recommended way of using the font, can we list this one last?

**Review by ericcornelissen — CHANGES_REQUESTED** [review]
Good start, nice work :+1: I have some initial comments mostly regarding the overall structure, let's see if we can work something out :slightly_smiling_face:

File: README.md
Comment (mondeja) [reply README.md:76]: Do you know how to get the directory where NPM serves all tarballs? :file_folder:
I have not found it anywhere...

File: README.md
Comment (mondeja) [reply README.md]: Better just `NodeJS` to avoid confusion with `Usage` section?

File: README.md
Comment (ericcornelissen) [reply README.md:63]: Hmm :thinking: We could add [automated GitHub releases](https://github.com/simple-icons/simple-icons/blob/18cac9da6f091444bc391467067183ab2336b0af/.github/workflows/publish.yml#L28-L73) and link to the latest instead? E.g.

```
https://github.com/simple-icons/simple-icons/releases/latest
```

Currently links to:


```
https://github.com/simple-icons/simple-icons/releases/tag/4.2.0
```


And is automatically updated each release.

File: README.md
Comment (ericcornelissen) [reply README.md]: Agreed! :+1:

File: README.md
Comment (ericcornelissen) [inline README.md:51]: I think an example will do to let people know how this works.

    Use any of the icons available in simple-icons by adding the following classes to a node in your HTML. Use the `simple-icons--color` class to apply the brand's color to the icon.

    ```html
    <i class="simpleicons simpleicons-[ICON NAME]"></i>
    <i class="simpleicons simpleicons-[ICON NAME] simple-icons--color"></i>
    ```

    Where `[ICON NAME]` is replaced by the icon name, for example:

    ```html
    <i class="simpleicons simpleicons-simpleicons"></i>
    <i class="simpleicons simpleicons-simpleicons simple-icons--color"></i>
    ```

    In this example we use the `<i>` tag, but any inline HTML tag should work as you expect.


We could still add more detailed documentation if it turns out to be necessary/you feel its necessary, but I would put it after the example. What do you think?

Also, I don't feel like it is necessary to explain that the `[ICON NAME]` should be the slug, at least we don't do it in the main project and it seems most people figure it out. That said, I'm not against adding it. Maybe wait and hope someone opens an issue if they experience difficulty because of this?

File: README.md
Comment (ericcornelissen) [inline README.md:58]: I guess using this badge is a bit nicer, mind pushing an update for this to `develop` for the main repo as well? (Or open a PR in case you think other maintainers should have a look at this first)

File: README.md
Comment (ericcornelissen) [inline README.md:20]: I think we should add a note here about the fact that this limits the scope to `v2` like we do in the main repository.

```suggestion
These examples use the latest major version. This means you won't receive any updates following the next major release. You can use `@latest` instead to receive updates indefinitely. However, this will result in a icons not showing as expected if the icon is removed.

### NodeJS
```

File: README.md
Comment (ericcornelissen) [inline README.md:16]: I think we should add the example for both JSDelivr and Unpkg, like in the main repo, agreed?

**Review by ericcornelissen — COMMENTED** [review]
Looks promising, got a bunch more comments, mostly to make it look more like the README of the main project :upside_down_face:

File: README.md
Comment (ericcornelissen) [reply README.md:16]: Also, wow this works I actually wasn't sure it would find the font files just like that :sweat_smile:

File: README.md
Comment (mondeja) [reply README.md:63]: Yes, would be the solution. I keep it like this and update when we add automated releases?

File: README.md
Comment (mondeja) [reply README.md:31]: OK. But I've used "Then, reference the CSS file" in the second to avoid repeat "You can" two times.

File: README.md
Comment (mondeja) [reply README.md:58]: I think we should avoid direct commits as much as possible to appear as transparent as possible. I'll upload a pull so someone could learn something about how to use shields.io, since they use simple-icons for icons.

File: README.md
Comment (mondeja) [reply README.md:20]: But only the NodeJS setup uses the latest major version with automatic updates not? Although the manual download gets the latest version, it's not updated automatically, so this warning should be inside "NodeJS" section :thinking:

File: README.md
Comment (mondeja) [reply README.md]: > I think we should add the example for both JSDelivr and Unpkg, like in the main repo, agreed?

Yes!

> Also, wow this works I actually wasn't sure it would find the font files just like that sweat_smile

CSS magic :sparkles: It seems that relative calls in `url()` are relative to the URL of file calling them, which it's very nice.

File: README.md
Comment (ericcornelissen) [reply README.md:24]: This note is intended for the CDN users. The comment isn't even correct for NPM/NodeJS users, as installing `@latest` will not give you updates indefinitely, only until the next major release (same as just `npm install simple-icon-font`). We could add a similar note to the NodeJS section as well, though I would expect a bit more knowledge in this regard from NPM users (but maybe that is unfair).

The reason, I think, we should add it here is because more often than not when I see packages offer CDN guidelines  in their docs the URL doesn't include a version, so the fact that we do include it may be unexpected to users of CDNs.

File: README.md
Comment (ericcornelissen) [inline README.md:22]: Shall we put them together?

```suggestion
```html
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/simple-icons-font@v2/font/simple-icons.min.css" type="text/css">
<link rel="stylesheet" href="https://unpkg.com/simple-icons-font@2/font/simple-icons.min.css" type="text/css">
```
```

File: README.md
Comment (ericcornelissen) [inline README.md:14]: Small typo here breaks the link to Unpkg

```suggestion
The font can be served from a CDN such as [JSDelivr][jsdelivr-link] or [Unpkg][unpkg-link]. Simply use the `simple-icons-font` npm package and specify a version in the URL like the following:
```

File: README.md
Comment (ericcornelissen) [inline README.md:32]: Was this a mistake in my suggestion? :fearful: 

```suggestion
After installation, the icon font and stylesheet can be found in `node_modules/simple-icons-font/font`. You can use your favorite bundling tool to include them into your project.
```

File: README.md
Comment (ericcornelissen) [inline README.md:64]: This example can be removed now, right?

```suggestion
```

**Review by ericcornelissen — COMMENTED** [review]
A couple more minor things (actually some things I got wrong in my previous review I think :sweat_smile:)

File: README.md
Comment (mondeja) [reply README.md:23]: I see. I've included changing only "if the icon is removed" by "if an icon is removed".

File: README.md
Comment (ericcornelissen) [reply README.md]: I'm 100% sure _"then"_ is unnecessary and 90% sure it is incorrect (though I'm not native in English either). This may be a bit easier to see if you simplify the sentence:

```diff
After installation, it can be found [...]
After installation, it can then be found [...]
```

File: README.md
Comment (ericcornelissen) [reply README.md]: I will open a PR to set up automated releases later, we can see in which PR we change it later :+1:

File: README.md
Comment (ericcornelissen) [reply README.md:23]: Good catch, forgot that removing one icon may lead to problems with other icons as well :+1:

Unfortunately I made another mistake and right now the sentence still isn't correct:

```diff
- However, this will result in a icons not showing as expected if an icon is removed.
+ However, this will result in icons not showing as expected if an icon is removed.
```

Comment (ericcornelissen) [reply README.md:23]: Can you fix this as well @mondeja?

Comment (mondeja) [reply README.md:23]: Well seen, it's done :+1:

Comment (ericcornelissen) [reply README.md:23]: > Good catch, forgot that removing one icon may lead to problems with other icons as well 👍

Hmm actually, this isn't true right? If an icon is removed from the main project, then other icons should still show up correctly when using CSS classes even though the exact unicode value changed?

If so, then the sentence as it is currently written can be a bit misleading. Currently it can legitemately be interpreted as meaning: if any single icon is removed from the main project, then all icons in the font can get messed up. By the above reasoning this is not true, Therefore, I think we should write something like:

> However, this will result in icons not showing as expected when they are removed.

or

> However, this will result in icons not showing as expected if removed.

**Comment by ericcornelissen** [comment]
@mondeja, any particular reason you didn't update the titles of the subsections of [Setup](https://github.com/mondeja/simple-icons-font/tree/setup-docs#setup) to "CDN Setup", "NodeJS Setup", and "Manual Setup"? I think a consistent style looks a bit better (and also "Setup from CDN" can be read as a sentence but is incorrect, which bothers me a bit 😅)

File: README.md
Comment (mondeja) [reply README.md:23]: What about this?

> However this may cause an icon to disappear if it has been removed in the latest version.

**Comment by mondeja** [comment]
> any particular reason you didn't update the titles of the subsections of Setup to "CDN Setup", "NodeJS Setup", and "Manual Setup"?

At least for me, it's redundant and very strange. This would make that we should raise the subsections a title level and eliminate the "Setup" section, right?

> and also "Setup from CDN" can be read as a sentence but is incorrect, which bothers me a bit

Yes, it's a bit weird. What about just "CDN"?

File: README.md
Comment (ericcornelissen) [reply README.md:23]: Also works :+1:

**Comment by ericcornelissen** [comment]
> > any particular reason you didn't update the titles of the subsections of Setup to "CDN Setup", "NodeJS Setup", and "Manual Setup"?
> 
> At least for me, it's redundant and very strange. This would make that we should raise the subsections a title level and eliminate the "Setup" section, right?


Fair enough, though I would personally disagree that it is "strange" and instead say it is just a stylistic choice. It arguably doesn't apply here, but if a book has a chapter Setup with several sections spanning multiple pages each of which talk about setup, it can be very helpful to repeat setup in each section title to remind the reader we're talking about the setup (making the assumption that readers typically finish sections and not chapters). Moreover, assuming everyone is going to go through the document top to bottom is, in my opinion, not a good assumption - I usually don't. That said, given the length of this document I would be fine leaving it out.

Also, this is exactly what the README in the main repo looks like: https://github.com/simple-icons/simple-icons#usage

**Comment by mondeja** [comment]
> Also, this is exactly what the README in the main repo looks like: https://github.com/simple-icons/simple-icons#usage

Oh, I didn't realize. Since in this case it is a matter of personal preferences, event if it's only to maintain consistency with the main repo, you've convinced me :+1:

**Comment by ericcornelissen** [comment]
Speaking about consistency, and this is my last comment I hope, the main repo uses ["Node Usage"](https://github.com/simple-icons/simple-icons#node-usage) and this draft currently has ["NodeJS Setup"](https://github.com/mondeja/simple-icons-font/tree/b480cb5498e0b58de92a70a0a9ff99c7b19e1e22#nodejs-setup) :sweat_smile:

**Comment by mondeja** [comment]
> Speaking about consistency, and this is my last comment I hope, the main repo uses "Node Usage" and this draft currently has "NodeJS Setup" :sweat_smile:

But the main repo includes how to use the API in that section and here the usage is common to all forms of setup.

File: README.md
Comment (mondeja) [reply README.md]: This will be unresolved until #54 merges. I think that we can wait for.

**Comment by ericcornelissen** [comment]
> But the main repo includes how to use the API in that section and here the usage is common to all forms of setup.

I meant that the difference is "NodeJS" vs. "Node", not sure how that relates? 🤔

**Comment by mondeja** [comment]
> I meant that the difference is "NodeJS" vs. "Node", not sure how that relates? :thinking:

Ah OK :sweat_smile: Sure, perhaphs "Node" would be cooler...

**Comment by ericcornelissen** [comment]
If you think NodeJS makes more sense that is fine, I just didn't understand [this comment](https://github.com/simple-icons/simple-icons-font/pull/53#issuecomment-748312506)

> But the main repo includes how to use the API in that section and here the usage is common to all forms of setup.

If that was with regards to "Usage" and "Setup" then my only argument for changing "NodeJS" into "Node" is consistency, but I would be fine with "NodeJS" as well. "NodeJS" might even be more clear than just "Node", though arguably "NPM" or "Package manager" is even better.

**Comment by mondeja** [comment]
> If you think NodeJS makes more sense that is fine, I just didn't understand this comment

I was confused thinking that you wanted to change the "Setup" word by "Usage" :sweat_smile: 

> "NodeJS" might even be more clear than just "Node", though arguably "NPM" or "Package manager" is even better.

I think so, but in this project perhaps keeping consistency with the main repository is something to take into account.

**Comment by ericcornelissen** [comment]
> I think so, but in this project perhaps keeping consistency with the main repository is something to take into account.

Okay, thanks for the clarification :slightly_smiling_face: 


> I think so, but in this project perhaps keeping consistency with the main repository is something to take into account.

Agreed, I like to keep it consistent as well. That does not mean it cannot be changed though. We can bring this discussion to the main repo to see if others agree. Once it's changed in the main repository we can change it here as well.

With that I will merge this as is :+1:

**Review by ericcornelissen — APPROVED** [review]

File: README.md
Comment (ericcornelissen) [reply README.md]: Sorry, didn't notice this comment until just after I merged this PR :sweat_smile: I will make sure to include this change in #54

**Pull request merged**
